### PR TITLE
Add deprecated `init` function with deprecated shape members

### PR DESCRIPTION
**Pull request by adam-fowler**
For shapes with deprecated member variables
- remove deprecated variables from original `init`.
- create a second `init` which includes the deprecated variables as parameters and flag this as deprecated.

### CODE REVIEW

**Comment by adam-fowler** [comment]
Forgot to get this reviewed

File: Sources/SotoCodeGenerator/AwsService+shapes.swift [hunk 1]
```diff
@@ -119,7 +119,8 @@ extension AwsService {
                     shapeOptions.append("allowStreaming")
                     if !payload.hasTrait(type: RequiresLengthTrait.self),
                        let operationShape = operationShape,
-                       operationShape.hasTrait(type: AwsAuthUnsignedPayloadTrait.self) {
+                       operationShape.hasTrait(type: AwsAuthUnsignedPayloadTrait.self)
+                    {
```
Comment (0xTim) [inline Sources/SotoCodeGenerator/AwsService+shapes.swift:123]: Was this meant to be changed?
```suggestion
                       operationShape.hasTrait(type: AwsAuthUnsignedPayloadTrait.self) {
```

**Review by 0xTim — CHANGES_REQUESTED** [review]
One formatting nit

File: Sources/SotoCodeGenerator/AwsService+shapes.swift (hunk 1 above)
Comment (0xTim) [reply Sources/SotoCodeGenerator/AwsService+shapes.swift:123]: Ah fair enough

**Review by 0xTim — APPROVED** [review]

**Pull request merged**